### PR TITLE
feat: Dialog component

### PR DESCRIPTION
**Pull request by ChrisBaidoo**
## Description

Create a dialog Component that can be used in other repos. 

## Checklist:

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request). If you're unsure about any of these, don't hesitate to leave a comment on this pull request!

- [x] I have read the gliff.ai Contribution Guide.
- [x] I have requested to **pull a branch** and not from main.
- [x] I have checked all commit message styles match the requested structure.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my own code.
- [ ] I have assigned **3 or less** reviewers.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] New database changes have been committed.
- [ ] If appropriate, I have bumped any version numbers.

### CODE REVIEW

**Review by joshuajames-smith — CHANGES_REQUESTED** [review]
Looking good! Just a couple small changes:

- [x] The green `topbar` has an outline. Can this curve around the border radius? It currently looks like it has a gap.
- [x] Could the `x` be aligned centre in the green `topbar`?

Screenshot 2022-04-28 at 12 56 59

**Review by joshuajames-smith — CHANGES_REQUESTED** [review]
Thanks for making the edits. It's now centred but visually that makes it feel like it sits lower than this. Could it be bumped up a couple pixels to see how this looks?

Screenshot 2022-04-29 at 13 03 21

File: src/Card.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ export function Card(props: Props): JSX.Element {
           <MaterialIconButton
             onClick={props.handleClose}
             size="small"
-            sx={{ position: "absolute", top: "10px", right: "7px" }}
+            sx={{ position: "absolute", top: "8px", right: "7px" }}
```
Comment (ChrisBaidoo) [inline src/Card.tsx:57]: Screenshot 2022-05-03 at 10 39 12

@joshuajames-smith let me know if this is any better. Thanks

**Review by joshuajames-smith — APPROVED** [review]

File: src/Card.tsx (hunk 1 above)
Comment (joshuajames-smith) [reply src/Card.tsx:57]: Great, thank you!

**Review by SilviaZeta — APPROVED** [review]

**Pull request merged**